sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -65,6 +65,43 @@ private[sql] class RocksDBStateStoreProvider
       value
     }
 
+    override def valuesIterator(key: UnsafeRow, colFamilyName: String): Iterator[UnsafeRow] = {
+      verify(key != null, "Key cannot be null")
+      verify(encoder.supportsMultipleValuesPerKey, "valuesIterator requires a encoder " +
+      "that supports multiple values for a single key.")
+      val valueIterator = encoder.decodeValues(rocksDB.get(encoder.encodeKey(key), colFamilyName))
+
+      if (!isValidated && valueIterator.nonEmpty) {
```

Comment:
hmm - why do we have this check ?

Reply:
Current logic for validation only tests the first key/value pair (I am not aware of the historical reasoning behind this). The code change just preserves this functionality.

Reply:
Should we just remove this ? cc - @HeartSaVioR

Reply:
historical reasoning - unfortunately we had two difference state store checker being proposed at the same time. This is to cover the case with broken UnsafeRow. I guess it's to cover the case of schema check with retaining the characteristic of schema-on-read (which is arguably replaced by state schema checker), but it still capture the case of incompatible UnsafeRow being broken in any way.

Maybe checking a single key-value pair is not sufficient in any way though.

Reply:
I'm OK to not add the check here.

Reply:
Removed the check. Thanks for the context.

Reply:
@HeartSaVioR - is it ok to remove this check everywhere ? I can create the change in a separate PR. Don't think this is giving us too much currently

Reply:
Discussed offline. There had been a couple of cases which inspection of state row in underneath were helpful on debugging. Maybe we could remove this once we improve state data source reader to produce state key-value as binary, but before that let's keep it as it is for existing codebase.